### PR TITLE
[WIP] Several improvements + handle OPTIONS query nginx

### PR DESCRIPTION
**Pull request by RodinVasily**
### Description

- Increasing request time
- Database dependency for GLUE
- CORS headers for Zed container
- https://spryker.atlassian.net/browse/AL2021-1081 - Preflight OPTIONS requests in Nginx

### Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: generator/src/templates/nginx/http/glue.server.conf.twig [hunk 1]
```diff
@@ -20,4 +28,18 @@
 {% if storeServices['mail']['sender']['name'] is not empty %}
         fastcgi_param SPRYKER_MAIL_SENDER_NAME "{{ storeServices['mail']['sender']['name'] }}";
 {% endif %}
+
+        if ($request_method = 'OPTIONS') {
+            add_header 'Access-Control-Allow-Origin' '$allow_origin' always;
+            add_header 'Access-Control-Allow-Methods' 'GET, POST, PUT, DELETE, PATCH, HEAD, OPTIONS' always;
+            add_header 'Access-Control-Allow-Headers' '{{ endpointData['cors-allow-headers'] | default('') | nginx_var }}' always;
```
Comment (alexanderM91) [inline generator/src/templates/nginx/http/glue.server.conf.twig:35]: Is it the custom one `cors-allow-headers`?

**Review by alexanderM91 — CHANGES_REQUESTED** [review]

File: generator/src/templates/nginx/http/glue.server.conf.twig (hunk 1 above)
Comment (RodinVasily) [reply generator/src/templates/nginx/http/glue.server.conf.twig:35]: yes, this is a custom field in `deploy.yml` file, just like `cors-allow-origin`

**Comment by alexanderM91** [comment]
Hi @RodinVasily, 
JFYI: GlueBackend support has been released.

File: generator/src/templates/application/glue.yml.twig [hunk 2]
```diff
@@ -3,6 +3,7 @@
     depends_on:
       - key_value_store
       - search
+      - database
```
[the remark this answers is not in the document]
Comment (profuel) [reply generator/src/templates/application/glue.yml.twig:6]: @alexanderM91 do you have any updates about aforementioned new approach?

File: generator/src/templates/nginx/http/zed.server.conf.twig [hunk 3]
```diff
@@ -15,6 +15,9 @@
         fastcgi_param SPRYKER_BE_HOST "{{ host }}";
         fastcgi_param SPRYKER_BE_PORT "{{ externalPort }}";
 
+{% if endpointData['cors-allow-origin'] is not empty %}
```
[the remark this answers is not in the document]
Comment (profuel) [reply generator/src/templates/nginx/http/zed.server.conf.twig:30]: this is/was a workaround to allow CORS for Zed container, which is used by Glue Backend,

File: generator/src/templates/application/glue.yml.twig (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexanderM91) [reply generator/src/templates/application/glue.yml.twig:6]: Hi @profuel, yes it was released and ready for usage from the DockerSDK side starting from https://github.com/spryker/docker-sdk/compare/1.43.0...1.44.0

**Pull request closed without merging**